doc/develop/api/api_conventions.rst

```diff
@@ -0,0 +1,49 @@
+.. _api_conventions.rst:
+
+API Conventions
+###############
+
+Zephyr provides a few programming interfaces that users can use for creating their applications.
+Zephyr's API space includes the following areas:
+- C programming interface (function prototypes, structures, and macros), provided in C header files
+- Configuration system (Kconfig)
+- Device driver model (Devicetree)
+
+Other areas, such as the build system (CMake), may also be seen as Zephyr's API. However, any area
+that is not explicitly listed herein is not considered Zephyr's API as of today.
+
+API classification
+******************
+The files that define the Zephyr APIs may contain symbols intended for different usage. The intended
+API usage is designated by the API class. Zephyr APIs are classified as:
+
+- private - these APIs are intended for use in-tree within boundary of a software component such as
+            driver or subsystem. Private APIs are not subject to :ref:`api_lifecycle`. Therefore can
+            be changed or removed at any time.
+- internal - a class of APIs that serve as the interface between two or more low level areas.
+- public - a class of APis intended for use in-tree and out-of-tree. The public APIs are subject to
```

Comment:
Please qualify the api lifecycle statement to those APIs defined in the Zephyr repository unless we are also going to apply it to APIs of depended-upon modules (e.g., TFM, mbedtls, mcuboot).

Reply:
Added a sentence to clarify.